### DISCUSSION TITLE
Add a Description on to User Groups

### DISCUSSION THREAD

**Discussion by kmeilander** [umbraco/Umbraco-CMS#14986] [Features and ideas]
On the user groups, it would be nice if there was a description field with the group to help user managers know what a user group is for.  This is especially useful for custom permission logic, where you are setting logic in the code so it's not apparent based on the settings on the group.   

This could work similar to Doc Type descriptions in that it could be set under the name, and shown wherever the user group is chosen.

**Comment by kmeilander** [comment]
I opened a PR for this change.  https://github.com/umbraco/Umbraco-CMS/pull/14988

#### How this works:

On the User Group Edit screen, there is a field for the description under the User Group Name to allow Users to set the description in the same way that Document Type Descriptions are managed.
![image](https://github.com/umbraco/Umbraco-CMS/assets/1120701/1dd28917-d2b1-4545-892f-5eb09480d6be)


Once the description is set, it shows on the cards when choosing the group on a user,  so the admin user can get more details around what that User Group is for.
![image](https://github.com/umbraco/Umbraco-CMS/assets/1120701/57648548-0eb1-4c4a-967a-f5f8e5cfbaa7)


Once a Group is Added to a User, it shows the description along with the group details so other Users can see what this Group is for.

![image](https://github.com/umbraco/Umbraco-CMS/assets/1120701/6cc011c3-88d4-4a5e-bbec-d7106c769f4b)

**Comment by nielslyngsoe** [comment]
I may as well add that I like the idea.

The reason is that we have more complex configuration options for User Groups, and from an overview perspective, seeing the allowed Sections and Start-Nodes does not necessarily provide a clear idea of what the User Group permits. So this will enable us to remove those specifics in the future.

And additionally, this enables spelling out the project-specific aspect of what this User Group covers. 👍

**Comment by AndyButland** [comment]
This has been addressed now in https://github.com/umbraco/Umbraco-CMS/pull/21057 so I'll close this discussion when the PR is merged.  Just wanted to say thanks to @kmeilander for both raising the suggestion and proposing a PR to add it.  Unfortunately it didn't make the cut for 13 but I'm glad to say we've got to it for 17, and the initial implementation was very helpful as reference.

**Comment by kmeilander** [comment]
It's great to hear this will get into v17.  Thanks all who helped make this happen!

**Comment by AndyButland** [comment]
Resolved via https://github.com/umbraco/Umbraco-CMS/pull/21057 for 17.2.
